### PR TITLE
[zh-cn]: Update translation of String.raw()

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#13975

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md [hunk 1]
```diff
@@ -31,42 +33,83 @@ String.raw`templateString`
...
 ## 描述
 
-在大多数情况下，`String.raw()` 是用来处理模版字符串的。不要被上面复杂的参数要求吓到，因为像所有的 [tag functions](/zh-CN/docs/Web/JavaScript/Reference/template_strings#Tagged_template_literals)一样，你通常不需要把它看成一个普通函数，你只需要把它放在模板字符串前面就可以了，而不是在它后面加个括号和一堆参数来调用它，引擎会替你去调用它。
+在大多数情况下，`String.raw()` 是用来处理模版字符串的。不要被上面复杂的参数要求吓到，因为像所有的[标签函数](/zh-CN/docs/Web/JavaScript/Reference/template_strings#带标签的模版)一样，你通常不需要把它看成一个普通函数，你只需要把它放在模板字符串前面就可以了，而不是在它后面加个括号和一堆参数来调用它，引擎会替你去调用它。
 
 `String.raw()` 是唯一一个内置的模板字符串标签函数，因为它太常用了。不过它并没有什么特殊能力，你自己也可以实现一个和它功能一模一样的标签函数。
 
+> **警告：** 不应直接使用 `String.raw` 作为“身份”标签。请参见[构建身份标签](#building_an_identity_tag)以了解如何实现此操作。
+
+如果使用没有 `length `属性或 `length` 为非正数的 `raw` 属性的对象调用 `String.raw()` 方法，它将返回一个空字符串 `""`。如果 `substitutions.length < strings.raw.length - 1`（即没有足够的替换项来填充占位符——这在一个格式正确的标记模板字面量中是不可能发生的），则其余的占位符将被填充为空字符串。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md:46]:
```suggestion
如果使用没有 `length` 属性或 `length` 为非正数的 `raw` 属性的对象调用 `String.raw()` 方法，它将返回一个空字符串 `""`。如果 `substitutions.length < strings.raw.length - 1`（即没有足够的替换项来填充占位符——这在一个格式正确的标记模板字面量中是不可能发生的），则其余的占位符将被填充为空字符串。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md [hunk 2]
```diff
@@ -79,6 +122,8 @@ String.raw(
 
 ## 参见
 
-- [模板字符串](/zh-CN/docs/Web/JavaScript/Reference/template_strings)
+- [`core-js` 中 `String.raw` 的 polyfill](https://github.com/zloirock/core-js#ecmascript-string-and-regexp)
+- [模版字符串](/zh-CN/docs/Web/JavaScript/Reference/Template_literals)
 - {{jsxref("String")}}
-- [JavaScript 词法](/zh-CN/docs/Web/JavaScript/Reference/Lexical_grammar)
+- [词法文法](/zh-CN/docs/Web/JavaScript/Reference/Lexical_grammar)
+-
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md:129]:
```suggestion

```

File: files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md [hunk 3]
```diff
@@ -79,6 +122,7 @@ String.raw(
 
 ## 参见
 
-- [模板字符串](/zh-CN/docs/Web/JavaScript/Reference/template_strings)
+- [`core-js` 中 `String.raw` 的 polyfill](https://github.com/zloirock/core-js#ecmascript-string-and-regexp)
+- [模版字符串](/zh-CN/docs/Web/JavaScript/Reference/Template_literals)
 - {{jsxref("String")}}
-- [JavaScript 词法](/zh-CN/docs/Web/JavaScript/Reference/Lexical_grammar)
+- [词法文法](/zh-CN/docs/Web/JavaScript/Reference/Lexical_grammar)
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md:128]:
```suggestion
- [词法文法](/zh-CN/docs/Web/JavaScript/Reference/Lexical_grammar)

```

File: files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md [hunk 4]
```diff
@@ -31,42 +33,83 @@ String.raw`templateString`
...
 
 ```js
 String.raw`Hi\n${2 + 3}!`;
-// 'Hi\\n5!'，Hi 后面的字符不是换行符，\ 和 n 是两个不同的字符
+// 'Hi\\n5!'，'Hi' 后面的字符不是换行符，'\' 和 'n' 是两个不同的字符。
 
 String.raw`Hi\u000A!`;
-// "Hi\\u000A!"，同上，这里得到的会是 \、u、0、0、0、A 6 个字符，
-// 任何类型的转义形式都会失效，保留原样输出，不信你试试.length
+// 'Hi\\u000A!'，同上，这里得到的会是 \, u, 0, 0, 0, A, 6 个字符，
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md:57]:
```suggestion
// 'Hi\\u000A!'，同上，这里得到的会是 \、u、0、0、0、A，6 个字符。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md [hunk 5]
```diff
@@ -31,42 +33,83 @@ String.raw`templateString`
...
+// "<canvas>\\n</canvas>"
+```
+
+对于“真实的标识”标签而言，这可能并不是你想要的，因为这种情况下标签纯粹用于标记化，不会改变字面量的值。在这种情况下，你可以创建一个自定义标签，并将“处理过”的字面量数组传递给 `String.raw`，假装它们是原始字符串。
+
+```js-nolint
+const html = (strings, ...values) => String.raw({ raw: strings }, ...values);
+// 一些格式化工具将会将这个字面量的内容格式化为 HTML
+const doc = html`<canvas>\n</canvas>`;
+// "<canvas>\n</canvas>"; the "\n" becomes a line break
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md:106]:
```suggestion
// "<canvas>\n</canvas>"; "\n" 变成了换行符
```

File: files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md [hunk 6]
```diff
@@ -31,42 +33,83 @@ String.raw`templateString`
...
+```
+
+请注意，第一个参数是一个带有 `raw` 属性的对象，其值是一个类数组对象（具有 `length` 属性和整数索引），表示模板字面量中分隔的字符串。其余的参数是替换项。由于 `raw` 值可以是任何类数组对象，它甚至可以是一个字符串！例如，`'test'` 被视为 `['t', 'e', 's', 't']`。以下代码等效于 `` `t${0}e${1}s${2}t` ``：
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md:109]:
```suggestion
请注意，第一个参数是一个带有 `raw` 属性的对象，其值是一个类数组对象（具有 `length` 属性和整数索引），表示模板字面量中分隔的字符串。其余的参数是替换项。由于 `raw` 值可以是任何类数组对象，它甚至可以是一个字符串！例如，`'test'` 被视为 `['t', 'e', 's', 't']`。以下代码与 `` `t${0}e${1}s${2}t` `` 等效：
```

File: files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md [hunk 7]
```diff
@@ -31,42 +33,83 @@ String.raw`templateString`
 ### 异常
 
 - {{jsxref("TypeError")}}
-  - : 如果第一个参数没有传入一个格式正确的对象，则会抛出 `TypeError` 异常。
+  - : 如果第一个参数没有 `raw` 属性，或者 `raw` 属性的值为 `undefined` 或 `null`，则会抛出异常。
 
 ## 描述
 
-在大多数情况下，`String.raw()` 是用来处理模版字符串的。不要被上面复杂的参数要求吓到，因为像所有的 [tag functions](/zh-CN/docs/Web/JavaScript/Reference/template_strings#Tagged_template_literals)一样，你通常不需要把它看成一个普通函数，你只需要把它放在模板字符串前面就可以了，而不是在它后面加个括号和一堆参数来调用它，引擎会替你去调用它。
+在大多数情况下，`String.raw()` 是用来处理模版字符串的。不要被上面复杂的参数要求吓到，因为像所有的[标签函数](/zh-CN/docs/Web/JavaScript/Reference/template_strings#带标签的模版)一样，你通常不需要把它看成一个普通函数，你只需要把它放在模板字符串前面就可以了，而不是在它后面加个括号和一堆参数来调用它，引擎会替你去调用它。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md:40]:
```suggestion
在大多数情况下，`String.raw()` 与模板字符串一起使用。上面提到的第一种语法很少使用，因为 JavaScript 引擎会自动为你调用它并传入适当的参数（就像其他[标记函数](/zh-CN/docs/Web/JavaScript/Reference/Template_literals#带标签的模版)一样）。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md [hunk 8]
```diff
@@ -31,42 +33,83 @@ String.raw`templateString`
...
+  - : 如果第一个参数没有 `raw` 属性，或者 `raw` 属性的值为 `undefined` 或 `null`，则会抛出异常。
 
 ## 描述
 
-在大多数情况下，`String.raw()` 是用来处理模版字符串的。不要被上面复杂的参数要求吓到，因为像所有的 [tag functions](/zh-CN/docs/Web/JavaScript/Reference/template_strings#Tagged_template_literals)一样，你通常不需要把它看成一个普通函数，你只需要把它放在模板字符串前面就可以了，而不是在它后面加个括号和一堆参数来调用它，引擎会替你去调用它。
+在大多数情况下，`String.raw()` 与模板字符串一起使用。上面提到的第一种语法很少使用，因为 JavaScript 引擎会自动为你调用它并传入适当的参数（就像其他[标记函数](/zh-CN/docs/Web/JavaScript/Reference/Template_literals#带标签的模版)一样）。
 
 `String.raw()` 是唯一一个内置的模板字符串标签函数，因为它太常用了。不过它并没有什么特殊能力，你自己也可以实现一个和它功能一模一样的标签函数。
 
+> **警告：** 不应直接将 `String.raw` 作为“标识”标签。请参见[构建身份标签](#构建身份标签)以了解如何实现此操作。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md:44]: Could you also update the rest of "identity tag" (to "标识标签").

File: files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md [hunk 9]
```diff
@@ -31,42 +33,83 @@ String.raw`templateString`
...
-// "Hi\nBob!"，内插表达式还可以正常运行
+// 'Hi\\nBob!'，内插表达式还可以正常替换。
+
+String.raw`Hi \${name}!`;
+// 'Hi \\${name}!'，美元符号被转义；这里没有插值。
+```
 
-// 正常情况下，你也许不需要将 String.raw() 当作函数调用。
-// 但是为了模拟 `t${0}e${1}s${2}t` 你可以这样做：
+### 构建身份标签
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md:69]: +1

File: files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md (hunk 8 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/string/raw/index.md:44]:
```suggestion
> **警告：** 不应直接将 `String.raw` 作为“标识”标签。请参见[构建标识标签](#构建标识标签)以了解如何实现此操作。
```

**Review by yin1999 — APPROVED** [review]
Thanks 🙏

**Pull request merged**